### PR TITLE
expose the quic.Config in h2quic.Server and h2quic.RoundTripper

### LINKED ISSUE #677: Expose quic.Config in h2quic.Server and h2quic.RoundTripper

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #677.

I made dependency injection a bit easier to increase testabilty of the h2quic package.

### CODE REVIEW

File: h2quic/client_test.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ var _ = Describe("Client", func() {
 	It("dials", func(done Done) {
 		client = newClient(nil, "localhost:1337", &roundTripperOpts{})
 		session.streamsToOpen = []quic.Stream{newMockStream(3), newMockStream(5)}
-		client.dialAddr = func(hostname string, _ *tls.Config, _ *quic.Config) (quic.Session, error) {
+		dialAddr = func(hostname string, _ *tls.Config, _ *quic.Config) (quic.Session, error) {
```
Comment (lucas-clemente) [inline h2quic/client_test.go]: The var should be reset to the original value after each test.

File: h2quic/server_test.go [hunk 2]
```diff
@@ -436,31 +440,13 @@ var _ = Describe("H2 server", func() {
...
-		Expect(ok).To(BeTrue())
-		if runtime.GOOS == "windows" {
-			// for some reason, Windows return a different error number, corresponding to an WSAEADDRINUSE error
-			// see https://msdn.microsoft.com/en-us/library/windows/desktop/ms681391(v=vs.85).aspx
-			Expect(syscallErr.Err).To(Equal(syscall.Errno(0x2740)))
-		} else {
-			Expect(syscallErr.Err).To(MatchError(syscall.EADDRINUSE))
+	It("errors when listening fails", func() {
+		testErr := errors.New("listen error")
+		quicListenAddr = func(addr string, tlsConf *tls.Config, config *quic.Config) (quic.Listener, error) {
```
Comment (lucas-clemente) [inline h2quic/server_test.go]: Should be reset to its original value after the test.

File: h2quic/client_test.go [hunk 3]
```diff
@@ -44,19 +46,31 @@ var _ = Describe("Client", func() {
...
+		client = newClient("quic.clemente.io", nil, &roundTripperOpts{}, nil)
 		Expect(client.hostname).To(Equal("quic.clemente.io:443"))
 	})
 
 	It("dials", func(done Done) {
-		client = newClient(nil, "localhost:1337", &roundTripperOpts{})
+		client = newClient("localhost:1337", nil, &roundTripperOpts{}, nil)
 		session.streamsToOpen = []quic.Stream{newMockStream(3), newMockStream(5)}
-		client.dialAddr = func(hostname string, _ *tls.Config, _ *quic.Config) (quic.Session, error) {
+		dialAddr = func(hostname string, _ *tls.Config, _ *quic.Config) (quic.Session, error) {
```
Comment (lucas-clemente) [inline h2quic/client_test.go]: Again, bc github hides it: The var should be reset to the original value after each test.

File: h2quic/server_test.go (hunk 2 above)
Comment (marten-seemann) [inline h2quic/server_test.go]: I think it already is. Look at the `AfterEach` in line 86.

File: h2quic/client_test.go (hunk 3 above)
Comment (marten-seemann) [inline h2quic/client_test.go]: Fixed.

File: h2quic/server_test.go (hunk 2 above)
Comment (lucas-clemente) [inline h2quic/server_test.go]: Right, missed that.

**Pull request merged**